packages/app-mobile/components/ExtendedWebView.tsx

```diff
@@ -0,0 +1,143 @@
+// Wraps react-native-webview. Allows loading HTML directly.
+
+import * as React from 'react';
+
+import {
+	forwardRef, Ref, useEffect, useImperativeHandle, useRef, useState,
+} from 'react';
+import { WebView, WebViewMessageEvent } from 'react-native-webview';
+import { WebViewErrorEvent, WebViewEvent, WebViewSource } from 'react-native-webview/lib/WebViewTypes';
+
+import Setting from '@joplin/lib/models/Setting';
+import { themeStyle } from '@joplin/lib/theme';
+import { Theme } from '@joplin/lib/themes/type';
+import shim from '@joplin/lib/shim';
+import { StyleProp, ViewStyle } from 'react-native';
+
+
+export interface WebViewControl {
+	// Evaluate the given [script] in the context of the page.
+	// Unlike react-native-webview/WebView, this does not need to return true.
+	injectJS(script: string): void;
+}
+
+interface SourceFileUpdateEvent {
+	uri: string;
+	baseUrl: string;
+
+	filePath: string;
+}
+
+type OnMessageCallback = (event: WebViewMessageEvent)=> void;
+type OnErrorCallback = (event: WebViewErrorEvent)=> void;
+type OnLoadEndCallback = (event: WebViewEvent)=> void;
+type OnFileUpdateCallback = (event: SourceFileUpdateEvent)=> void;
+
+interface Props {
+	themeId: number;
+
+	// [html] should only be [null] if [html] is still being generated.
+	html: string|null;
```

Comment:
There's something off with this. Why does the calling code need to set the html prop to null? Can't it set it to an empty string? If it's empty you can still have a special case `setSource(undefined)`

Reply:
What I mean is that if I use your component and know nothing about it - why would I even set "html" to null? I'd set it to an empty string maybe if not everything is loaded, but not to "null" (or "undefined"). Yet your component seems to have a special behaviour for this. (whether it's "null" or "undefined" doesn't really matter)

What would happen if you change to just `html: string` and throw an error if the parent component set this to anything other than a string. Would something not work?